### PR TITLE
Handle Message Delivery logging and processing

### PR DESCRIPTION
**Pull request by influx6**
PR is intended to add the following features into the project.

- [X] Structured Logging with new log pkg
- [X] Have log emit delivery data into log for processing.
- [X] Have config provide option for setting message log file name and parameters.

Related to issue #325

### CODE REVIEW

File: cmd/statusd/library.go [hunk 1]
```diff
@@ -229,7 +229,7 @@ func CompleteTransaction(id, password *C.char) *C.char {
 	}
 	outBytes, err := json.Marshal(&out)
 	if err != nil {
-		log.Error("failed to marshal CompleteTransaction output", "error", err.Error())
+		log.Send(log.Errorf("failed to marshal CompleteTransaction output").With("error", err.Error()))
```
Comment (influx6) [inline cmd/statusd/library.go]: Should we swap usage like:

```log.Send(log.Errorf("failed to marshal CompleteTransaction output").With("error", err))```

to 

```log.Error("failed to marshal CompleteTransaction output", log.Field{"error": err})``` ?

File: cmd/statusd/main.go [hunk 2]
```diff
@@ -85,7 +85,7 @@ var (
 	// LogLevelFlag defines a log reporting level
 	LogLevelFlag = cli.StringFlag{
 		Name:  "log",
-		Usage: `Log level, one of: "ERROR", "WARN", "INFO", "DEBUG", and "TRACE"`,
+		Usage: `Log level, one of: "ERROR", "YELLOWALERT", "INFO", "REDALERT"`,
```
Comment (influx6) [inline cmd/statusd/main.go]: @tiabc  @divan I believe this guys suite more for the purpose we want both from discussions from #295 and the fact that i believe `Trace` should naturally be part of log entries, where each has `file` and `line number` associated with area of called function.

File: geth/log/custom/custom.go [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+	"github.com/fatih/color"
+	"github.com/status-im/status-go/geth/log"
+)
+
+// FlatDisplay writes giving Entries as seperated blocks of contents where the each content is
+// converted within a block like below:
+//
+//  Message: We must create new standard behaviour 	Function: BuildPack  |  display: red,  words: 20,
+//
+//  Message: We must create new standard behaviour 	Function: BuildPack  |  display: red,  words: 20,
```
Comment (influx6) [inline geth/log/custom/custom.go]: @adambabik This is how current log lines are displayed:

```Message: RPC Message Status Notification  Function: github.com/status-im/status-go/geth/node.(*NodeManager).startNode.func1  File: /Users/darkvoid/devlabs/src/github.com/status-im/status-go/geth/node/manager.go:99  |  STATUSIM_MESSAGE_STATE_NOTIFICATION: {"reason":null,"envelope":{"Version":"AA==","Expiry":1508427178,"TTL":20,"Topic":"0x796000e9","AESNonce":null,"Data":"BErMAvKCU4BZiAsmg226YE/WYFbnmw4C1fcvKpsUBS7jNYAJJJ3IGH/sQC+hiO3goIA0Rhq/akX7IMbBqlQWEowcV1xW6aNZmkibh6+zsyvjvqzq06s6cXolTmLEuaNwhDg3gcgn6vXQagMziPN3gb6DrfDHd7bB/2pEeoTcGBPoJHYphlFBGtS3ZitBEtCB5qJ6q+1g9GX4p/4Qen3J56pO7yd8RwGf1vA8Nxj6AQeOMp+iguyFyWczUw31Edx57Vkl8j/V0ZjhTehfOoZ7VZs7m9icsFsgrU0tbtY779boIOtnvwSVRRZn40IGb6bD3ld/MiMhJZ+/6Ct9KmsHUpgRdFLB6jPVLkjAsXq+Kz/nfnuHa7RXoN4gh9WXbaRyNlJK1h124f9tOsPyhIx/Tbe2dFfvbyPaJU1r1B1TNUl/m7nIuBWL5XDBTJ8FxOC+2F4ThVwkOq0dAKn+QxDGA6N7p+KL1Fc5zVvDksGAODKR","EnvNonce":76},"timestamp":"2017-10-19T16:32:38.461709794+01:00","Source":{"symKeyID":"","pubKey":"0x","sig":"","ttl":0,"topic":"0x00000000","payload":"0x","padding":"0x","powTime":0,"powTarget":0,"targetPeer":""},"status":8,"direction":0,"received":null}  networkID: 3  message_direction: UnknownDirection```

image

**Comment by influx6** [comment]
@tiabc Facing a odd behaviour with CI:

- In CI:

image

On console with same command:

image

File: cmd/statusd/library.go (hunk 1 above)
Comment (adambabik) [reply cmd/statusd/library.go]: I have been thinking about not changing the `log` API at all. It already has a good format:
```
log.Error("message", "tag1", value1, "tag2", value2, ...)
```
I believe you can convert it into JSON using a custom handler.

Comment (influx6) [reply cmd/statusd/library.go]: Ya, spoke to @tiabc at length yesterday, will be reverting the log package back, and only keep changes in `goethereum` in `vendor`.

**Comment by influx6** [comment]
Because of the massive amount of changes in this PR, i will be closing it. WIll be making a PR specifically for just logging delivery status, but I would be keeping the `DeliveryService` system, i think being able to plug into `ethereum` from status, is a far more flexible and suitable approach.

**Pull request closed without merging**